proposals/utils/pdf_diff_logic.py

```diff
@@ -0,0 +1,1132 @@
+        'amount_per_week',
+        'duration',
+        'measurement',
+        'experimenter',
+        'description',
+        'has_controls',
+        'controls_description',
+        'extra_task'
+    ]
+
+    def get_row_fields(self, obj):
+        rows = copy(self.row_fields)
+
+        if obj.version == 1:
+            fields_to_remove = ['multiple_sessions',
+                                'session_frequency',
+                                'extra_task']
+            for field in fields_to_remove:
+                rows.remove(field)
+        else:
+            rows.remove('amount_per_week')
+            if not obj.multiple_sessions:
+                rows.remove('session_frequency')
+            if obj.settings_contains_schools:
+                rows.remove('extra_task')
+        
+        if not needs_details(obj.setting.all()):
+            rows.remove('setting_details')
+        if not obj.study.has_children() or \
+        not needs_details(obj.setting.all(), 'needs_supervision'):
+            rows.remove('supervision')
+            rows.remove('leader_has_coc')
+        elif obj.supervision:
+            rows.remove('leader_has_coc')
+        if not obj.has_controls:
+            rows.remove('controls_description')   
+
+        return rows
+    
+    def render(self, context):
+        if self.object.study.proposal.studies_number > 1:
+            context.update(
+                {
+                    'study_title': super().get_study_title(self.object.study)
+                }
+            )
+        return super().render(context)
+    
+class InterventionSectionDiff(DiffSection, InterventionSection):
+
+    def render(self, context):
+        if self.object.study.proposal.studies_number > 1:
+            context.update(
+                {
+                    'study_title': super().get_study_title(self.object.study)
+                }
+            )
+        return super().render(context)
+    
+class ObservationSection(InterventionSection):
+    '''This class receives an observation object
+    It inherits from the InterventionSection, as it can repurpose the same overridden
+    render method.'''
+    section_title = _('Het observatieonderzoek')
+    row_fields = [
+        'setting',
+        'setting_details',
+        'supervision',
+        'leader_has_coc',
+        'days',
+        'mean_hours',
+        'details_who',
+        'details_why',
+        'details_frequency',
+        'is_anonymous',
+        'is_anonymous_details',
+        'is_in_target_group',
+        'is_in_target_group_details',
+        'is_nonpublic_space',
+        'is_nonpublic_space_details',
+        'has_advanced_consent',
+        'has_advanced_consent_details',
+        'needs_approval',
+        'approval_institution',
+        'approval_document',
+        'registrations',
+        'registrations_details'        
+    ]
+
+    def get_row_fields(self, obj):
+        rows = copy(self.row_fields)
+
+        if obj.version == 1:
```

Comment:
It might make more sense to make a `v1` and a `v2` specific class (with a common base class or mixin for the stuff they share) and do the version check in `create_context_pdf`.

This will allow you to remove the 'remove these fields in case of version x' code and just not have those fields in `row_fields`.
In the future we could then make a new class for a possible `v3`, instead of adding an extra `if` here (making the class less clear).

Reply:
I started working on this, but did not really find it that much more elegant ... If you really want, I will implement this, but creating a 2 extra baseclasses, 2 more version classes, plus more logic in the create context functions, to me, makes it more complicated.

Reply:
Yeah fair; at this point it's probably a bit unnecessary. It was a bit of a 'that might be neat in the future' suggestion, but it's fair to say we can defer this to if and when that future arrives